### PR TITLE
Migrated errors for models package #2

### PR DESCRIPTION
**Pull request by dhruv0000**
**Description**

This PR fixes:
#3485 
#3486 
#3487 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/error.go [hunk 1]
```diff
@@ -76,6 +107,42 @@ func ErrUnmarshal(err error, obj string) error {
...
+
+func ErrDelete(err error, obj string, statusCode int) error {
+	return errors.New(ErrDeleteCode, errors.Alert, []string{"Unable to delete data from the Provider", obj}, []string{"Status Code: " + fmt.Sprint(statusCode), err.Error()}, []string{}, []string{})
+}
+
+func ErrDecodeBase64(err error, obj string) error {
+	return errors.New(ErrDecodeBase64Code, errors.Alert, []string{"Error occurred while decoding base65 string", obj}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrMarshalPKIX(err error) error {
```
Comment (Shreyas220) [inline models/error.go:130]: Everything looks great, but just one question we already have ErrMarshal() 
Was there a specific reason to create this function ?

Comment (kumarabd) [reply models/error.go:130]: @Shreyas220 great catch. @dhruv0000 We will want to reuse the Marshal function.

Comment (dhruv0000) [reply models/error.go:130]: yes. I did thought of it, but this error is specifically for converting a public key to PKIX (X.509) format and not JSON, so I thought it deserved its own error type.

Comment (kumarabd) [reply models/error.go:130]: If the statement and everything is the same, then irrespective of the use case we could re-use.

**Pull request merged**